runtime/jvmti/suspendhelper.cpp

```diff
@@ -45,7 +45,14 @@ suspendThread(J9VMThread *currentThread, jthread thread, UDATA allowNull, UDATA
 				if (currentThread == targetThread) {
 					*currentThreadSuspended = TRUE;
 				} else {
+					/* It is a well known problem that suspending a Java thread may cause deadlocks in the application if a resource is 
+					 * held while suspended. To prevent this happening in the Thread JCL library, check that the Object java.lang.Thread.lock 
+					 * has been released before suspending. To make sure of this, acquire the lock while halt is in progress.
+					 */
+					vm->internalVMFunctions->objectMonitorEnter(currentThread, J9VMJAVALANGTHREAD_LOCK(targetThread, targetThread->threadObject));
```

Comment:
Please cache the internal function table in a local (and change the other uses in this function to use the local).

https://github.com/eclipse/openj9/blob/566007db997abe3f47f25565506c0f5a0a2f1b00/runtime/j9vm/java11vmi.c#L319

Reply:
updated.